### DISCUSSION TITLE
Segmentation fault (core dumped)

### DISCUSSION THREAD

**Discussion by liz109** [llnl/LEAP#65] [Q&A] [answered]
Hello,

I am trying to generate curved projections following the demo d09_curved_polygon_detector_array.py. The raw ct images with shape (nz, 512, 512). Using the default geometry configuration, I got the reconstructed images in 100 angles like figure 1.  But when I changed the geometry, e.g. numSticks=64 or numCols_per_module=32, I got Error: Segmentation fault (core dumped). Also, could you please give me some suggestions to enlarge the FOV? Thank you.
![figure1](https://github.com/LLNL/LEAP/assets/43913666/b3de03b8-1499-42a6-bf17-b9fdf0e396b6)

**Comment by kylechampley** [comment, marked as answer]
That demo script is for when your raw projection data is composed as an arc of several modules. Is this really what you want or is your data just cone-beam with a curved array?

If it is just a simple curved array, just follow the first demo script but run this command:
leapct.set_curvedDetector()

To enlarge the FOV, just define a bigger reconstruction volume and to enlarge the circular mask use this function;
leapct.set_diameterFOV(d)

https://leapct.readthedocs.io/en/latest/ctvolume.html#leapctype.tomographicModels.set_diameterFOV

**Comment by kylechampley** [comment]
I assume this fixed your problem, so I am going to close this discussion.  Feel free to reopen if further issues arise.
